### PR TITLE
Create reusable CartItem component

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Create a reusable `CartItem` component that can be used also outside of `CartInventory`

## Justify why they are needed

- Based on the design system; that's why I call it `CartItem`.

- Accept children to allow custom actions based on the context.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2738** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2738" target="_blank">Graphite</a> 
    * **PR #2739** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2739" target="_blank">Graphite</a>  👈
      * **PR #2740** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2740" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2739?utm_source=stack-comment).

File: apps/store/src/components/CartItem/CartItem.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  productName: string
+  cost: ProductOfferFragment['cost']
+  documents: ProductOfferFragment['variant']['documents']
+  data: ProductOfferFragment['priceIntentData']
+
+  defaultExpanded?: boolean
+}
+
+export const CartItem = (props: Props) => {
+  const [expanded, setExpanded] = useState(props.defaultExpanded ?? false)
```
Comment (robinandeer) [inline apps/store/src/components/CartItem/CartItem.tsx:25]: Also aligned with states from UI kit

File: apps/store/src/components/CartItem/DetailsSheet.tsx [hunk 2]
```diff
@@ -4,13 +4,17 @@ import { useTranslation } from 'next-i18next'
...
-import { type DataTableRow, getDataTable } from '../DataTable/DataTable'
+import { type DataTableRow, getDataTable } from './DataTable/DataTable'
 
-type Props = Pick<
-  CartEntry,
-  'documents' | 'productName' | 'data' | 'tierLevelDisplayName' | 'deductibleDisplayName'
->
+type UserData = Record<string, unknown>
+
+type Props = {
```
Comment (robinandeer) [inline apps/store/src/components/CartItem/DetailsSheet.tsx:11]: Decouple from API

**Review by gustaveen — APPROVED** [review]
Sweet!
Added via Giphy

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2739).

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2739) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2739).

**Pull request merged**